### PR TITLE
Added flag to use kubelogin in non-admin user scenario and updated readme to reflect changes

### PR DESCRIPTION
**Pull request by tbarnes94**
In response to: https://github.com/Azure/aks-set-context/issues/25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +66,35 @@ You must run [Azure/login](https://github.com/Azure/login) before this action.
     cluster-name: "<cluster name>"
 ```
 
+### Non Admin User Example
+
```
Comment (OliverMKing) [inline README.md:70]: also explain that they need to install kubelogin and give an example

File: src/run.ts [hunk 2]
```diff
@@ -42,7 +45,19 @@ export async function run() {
   if (admin) cmd.push("--admin");
 
   const exitCode = await exec.exec(AZ_TOOL_NAME, cmd);
-  if (exitCode !== 0) throw Error("Az cli exited with error code " + exitCode);
+  if (exitCode !== 0) throw Error("az cli exited with error code " + exitCode);
+
+  if (useKubeLogin) {
+    const nonAdminCmd = [
```
Comment (OliverMKing) [inline src/run.ts:51]: rename to kubeloginCmd

File: src/run.ts [hunk 3]
```diff
@@ -42,7 +45,19 @@ export async function run() {
...
+  if (exitCode !== 0) throw Error("az cli exited with error code " + exitCode);
+
+  if (useKubeLogin) {
+    const nonAdminCmd = [
+      "convert-kubeconfig",
+      "-l",
+      "azurecli",
+    ]
+
+    const exitCode2 = await exec.exec(KUBELOGIN_TOOL_NAME, nonAdminCmd);
```
Comment (OliverMKing) [inline src/run.ts:57]: rename to kubeloginExitCode

**Review by OliverMKing — COMMENTED** [review]
Also need to update action.yml

File: src/run.ts (hunk 3 above)
Comment (tbarnes94) [reply src/run.ts:57]: done

File: src/run.ts (hunk 2 above)
Comment (tbarnes94) [reply src/run.ts:51]: done

File: README.md (hunk 1 above)
Comment (tbarnes94) [reply README.md:84]: done

**Comment by OliverMKing** [comment]
Still need to update action.yml

File: src/run.test.ts [hunk 4]
```diff
@@ -21,15 +21,15 @@ describe("Set context", () => {
...
+  }, 17500);
 
   it("throws without az tools", async () => {
     jest.spyOn(core, "getInput").mockImplementation((inputName, options) => {
       if (inputName == "resource-group") return resourceGroup;
       if (inputName == "cluster-name") return clusterName;
     });
     await expect(run()).rejects.toThrow();
-  });
+  }, 17500);
```
Comment (OliverMKing) [inline src/run.test.ts:32]: can you extract 17500 to a constant and maybe leave a comment with some context about the GitHub runner acting up?

Comment (tbarnes94) [reply src/run.test.ts]: done

File: README.md [hunk 5]
```diff
@@ -62,6 +66,50 @@ You must run [Azure/login](https://github.com/Azure/login) before this action.
...
+  with:
+    client-id: ${{ secrets.AZURE_CLIENT_ID }}
+    tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+    subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+- uses: azure/aks-set-context@v2.0
+  with:
+    resource-group: "<resource group name>"
+    cluster-name: "<cluster name>"
+    use-kubelogin: "true"
```
Comment (OliverMKing) [inline README.md:98]: they'd also need to specify that they are a non-admin user too?

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]

**Pull request merged**